Updating the default UI result

SparshithNR opened the pull request:
- Display a summary of the audit
- Provide an option to see the detailed view to see version violating the policy.
- Sample:
>Unsupported

<img width="677" alt="Screen Shot 2021-02-16 at 11 08 08 AM" src="https://user-images.githubusercontent.com/12808214/108121659-422a7600-7047-11eb-838e-e483cee83618.png">

> Unsupported with --verbose
 <img width="645" alt="Screen Shot 2021-02-16 at 11 07 28 AM" src="https://user-images.githubusercontent.com/12808214/108121611-2a52f200-7047-11eb-88bc-9f1e1c105c37.png">

> Supported 
<img width="699" alt="Screen Shot 2021-02-16 at 11 08 44 AM" src="https://user-images.githubusercontent.com/12808214/108121718-579fa000-7047-11eb-86a0-7541792c8b4c.png">

Commit:
Support node and ember-cli lts

Commit:
Fixing Linting issues

Commit:
Display in table format

Commit:
Showing more info

Commit:
so far

Commit:
Working feature

Commit:
Fully working

Commit:
Updating the CLI result sturcture

Commit:
Fixing the UI issue

suchitadoshi1987 commented on lib/help.js:
```diff
@@ -11,6 +11,7 @@ module.exports = chalk`
   {bold Options}
     {cyan --help, -h} this output
     {cyan --json, -h} output as json
+    {cyan --all} outputs detailed report of support audit
```
I would recommend this to be 
`--verbose/-v`
OR
`--details/-d`

SparshithNR replied:
`--verbose` makes sense and meets with a lot of conventions that already exist.

suchitadoshi1987 commented on lib/lts/index.js:
```diff
@@ -101,13 +101,32 @@ function ltsVersions(_time, currentDate) {
     });
 }
 
+function getCurrentLts(ltsList) {
```
please add some documentation to explain what this function does

suchitadoshi1987 commented on lib/lts/index.js:
```diff
@@ -101,13 +101,32 @@ function ltsVersions(_time, currentDate) {
+  let latestLts = '';
+  Object.keys(ltsList).forEach(version => {
+    const versionInfo = ltsList[version];
+    const today = new Date();
```
lets move this before `forEach`

suchitadoshi1987 commented on lib/lts/index.js:
```diff
@@ -101,13 +101,32 @@ function ltsVersions(_time, currentDate) {
+      latestLts = versionInfo.versionRange;
+    }
+  });
+  if (latestLts) {
```
is the aim for this function to short circuit and return if the `latestLTS` is found?

SparshithNR replied:
This function returns the latest LTS from the list we have in ember-cli-lts.json or node-lts.json. 
It also helps in throwing an exception when the above JSON file is not updated with the latest LTS.

Commit:
Fixing review comments
- change all to verbose
- added test when the LTS file is not updated
- update help
- add documentation for function

Commit:
Fixing broken test
Remove unused packages

Commit:
Fix windows test fails

Commit:
Adding behind column in the verbose

Avoid standard -v option for verbose alias.

Commit:
Adding feature to find expiring soon projects

suchitadoshi1987 commented on lib/lts/index.js:
```diff
@@ -101,13 +101,43 @@ function ltsVersions(_time, currentDate) {
+/**
+ *
+ * @param {list of LTS version} ltsList
+ * Returns current LTS version from the list of all LTS available
```
```suggestion
 * Returns current LTS version from the sorted list (old to new) of all available LTS versions.
```

SparshithNR replied:
This summary is wrong. This function returns the latest LTS from the list of all LTS available.

suchitadoshi1987 commented on lib/output/cli-output.js:
```diff
@@ -0,0 +1,256 @@
+  return table.toString();
+}
+
+/* Keep the order of voilations summary output as
```
```suggestion
/* Keep the order of violations summary output as
```

SparshithNR replied:
This change is not present in the latest

suchitadoshi1987 commented on lib/output/messages.js:
```diff
@@ -0,0 +1,53 @@
+const chalk = require('chalk');
+const { supportedRanges } = require('../time/index');
+
+const MILLSINQUARTER = 91 * 24 * 60 * 60 * 1000;
```
Please add a comment on the definition of this formula

suchitadoshi1987 commented on lib/project/index.js:
```diff
@@ -113,7 +121,12 @@ module.exports = async function isInSupportWindow(projectRoot) {
+            ...result,
+            name,
+            resolvedVersion,
+            latestVersion: info['dist-tags'].latest,
```
will the property `dist-tags` always be present in the `info` object?

SparshithNR replied:
Yeah, It must be present always.

Commit:
Fix singular/plural words

Commit:
Adding inline documentation for function

Commit:
separate each type of package details

Commit:
Add option to give only required type packages

Commit:
Adding tests and alias

stefanpenner committed:
Merge pull request #3 from SparshithNR/lts-support

Support node and ember-cli LTS versions

Commit:
All format fixing

Commit:
Move sort to a function and test it
